fastlane/fastlane
# [match] add git private key support for match

### Checklist

### Motivation and Context


We plan to have our `certificates` repository hosted GitHub and need to have access to it from different CI runners e.g. GitLab, Azure via [Deploy keys](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys).

### Description


I have added a new option to `match` called `git_private_key` and made it a conflicting option with `git_basic_authorization` and `git_bearer_authorization` and the other way around as well.

When `match` is used with this new option then the git command is wrapped in `ssh-agent bash -c 'ssh-add /path/given/id_dsa; GIT_COMMAND'`, this spawns a ssh-agent bash shell and adds private key to the ssh-agent. I learn this from [Martin v. Löwis@StackOverflow](https://stackoverflow.com/a/4565746/342437).

### Testing Steps


1. Have a certificates repository
2. Ensure that `ssh-add -l` does not list any identities related to the repository, to clear it `ssh-add -d IDENTITY` or simply `ssh-add -D`.
3. Now run `match` without the option `git_private_key` and it should fail.
4. Add `git_private_key` option to `match` and see it succeed.
5. Replace the path to a private key with `File.read(private_key_path)` and see it succeed.
